### ISSUE TITLE
Kilosort2 crashing and restarting computer

### ISSUE THREAD

**Issue by johnhwen1** [MouseLand/Kilosort2#75]
Hi all,

I've recently come across an issue where kilosort2 crashes and restarts the computer on some of my neuropixels recordings. Interestingly, it seems that I can get away with sorting just a portion of the problematic files (e.g., 500s out of a total 4000s recording). I was able to generate a Matlab crash dump file in one instance before the computer restarted. The file is copied below. Has anyone else seen anything like this before or have ideas on how to troubleshoot it?

Thanks in advance!

Best,
John

--------------------------------------------------------------------------------
          Access violation detected at Fri Jun 07 03:26:23 2019 -0700
--------------------------------------------------------------------------------

Configuration:
  Crash Decoding           : Disabled - No sandbox or build area path
  Crash Mode               : continue (default)
  Default Encoding         : windows-1252
  Deployed                 : false
  Graphics Driver          : NVIDIA Corporation GeForce GTX 1080/PCIe/SSE2 Version 4.6.0 NVIDIA 430.86
  Graphics card 1          : Intel Corporation ( 0x8086 ) Intel(R) UHD Graphics 630 Version 24.20.100.6290 (2018-8-27)
  Graphics card 2          : NVIDIA ( 0x10de ) NVIDIA GeForce GTX 1080 Version 26.21.14.3086 (2019-5-22)
  Java Version             : Java 1.8.0_152-b16 with Oracle Corporation Java HotSpot(TM) 64-Bit Server VM mixed mode
  MATLAB Architecture      : win64
  MATLAB Entitlement ID    : 1355671
  MATLAB Root              : C:\MATLAB\R2018b
  MATLAB Version           : 9.5.0.944444 (R2018b)
  OpenGL                   : hardware
  Operating System         : Microsoft Windows 10 Pro
  Process ID               : 2092
  Processor ID             : x86 Family 6 Model 158 Stepping 10, GenuineIntel
  Session Key              : 8c9d4663-4d1a-4685-ad36-6248dbb72112
  Window System            : Version 10.0 (Build 17763)

Fault Count: 1

Abnormal termination

Register State (from fault):
  RAX = 0000000000000000  RBX = 00000000043f7300
  RCX = 0000000000000000  RDX = 00000000c68de640
  RSP = 00000000043f6c90  RBP = 00000000043f6d90
  RSI = 0000000000000000  RDI = 0000000000000000
 
   R8 = 00000002987c1670   R9 = 00000000043f7300
  R10 = 000000000000001f  R11 = 00000000043f7168
  R12 = 0000000000000000  R13 = 00000001af9ba750
  R14 = 0000000000000000  R15 = 00000000043f7230
 
  RIP = 00007ff8636e2219  EFL = 00010206
 
   CS = 0033   FS = 0053   GS = 002b

Stack Trace (from fault):
[  0] 0x00007ff8636e2219             C:\code\KiloSort2\CUDA\mexMPnu8.mexw64+00008729 mexFunction+00000089
[  1] 0x00000000fc635524                               bin\win64\libmex.dll+00349476 MexRetrieveVersion+00003348
[  2] 0x00000000fc63571c                               bin\win64\libmex.dll+00349980 MexRetrieveVersion+00003852
[  3] 0x00000000fc635884                               bin\win64\libmex.dll+00350340 MexRetrieveVersion+00004212
[  4] 0x00000000fc619059                               bin\win64\libmex.dll+00233561 mexUnlock_800+00025273
[  5] 0x00000000170ae007                     bin\win64\pgo\m_dispatcher.dll+00057351 Mfh_file::dispatch_fh_impl+00001111
[  6] 0x00000000170ada9e                     bin\win64\pgo\m_dispatcher.dll+00055966 Mfh_file::dispatch_fh+00000062
[  7] 0x0000000017da4ead                            bin\win64\pgo\m_lxe.dll+00347821
[  8] 0x0000000017f3e9b6                            bin\win64\pgo\m_lxe.dll+02025910 MathWorks::lxe::ShutdownLxeEngine+00004034
[  9] 0x0000000017e9fd3c                            bin\win64\pgo\m_lxe.dll+01375548 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00493568
[ 10] 0x0000000017ea091c                            bin\win64\pgo\m_lxe.dll+01378588 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00496608
[ 11] 0x0000000017ea1c92                            bin\win64\pgo\m_lxe.dll+01383570 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00501590
[ 12] 0x0000000017ea28f8                            bin\win64\pgo\m_lxe.dll+01386744 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00504764
[ 13] 0x0000000017ea1ddf                            bin\win64\pgo\m_lxe.dll+01383903 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00501923
[ 14] 0x0000000017ea1ede                            bin\win64\pgo\m_lxe.dll+01384158 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00502178
[ 15] 0x0000000017da9a7d                            bin\win64\pgo\m_lxe.dll+00367229
[ 16] 0x0000000017dbb265                            bin\win64\pgo\m_lxe.dll+00438885
[ 17] 0x0000000017dba88c                            bin\win64\pgo\m_lxe.dll+00436364
[ 18] 0x0000000017db42a1                            bin\win64\pgo\m_lxe.dll+00410273
[ 19] 0x0000000017db39c6                            bin\win64\pgo\m_lxe.dll+00408006
[ 20] 0x0000000017db3ace                            bin\win64\pgo\m_lxe.dll+00408270
[ 21] 0x0000000017f93780                            bin\win64\pgo\m_lxe.dll+02373504 mwboost::archive::detail::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>+00012824
[ 22] 0x0000000017f9372e                            bin\win64\pgo\m_lxe.dll+02373422 mwboost::archive::detail::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>+00012742
[ 23] 0x0000000017e73623                            bin\win64\pgo\m_lxe.dll+01193507 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00311527
[ 24] 0x00000000197e9884                            bin\win64\pgo\m_lxe.dll+27891844 foundation::core::except::Exception<MathWorks::lxe::profiler::CoderProfilingException,std::exception,void>::what+00154880
[ 25] 0x00000000fb606bef                          bin\win64\libmwbridge.dll+00093167 MLInitialize_libmwbridge+00030799
[ 26] 0x00000000197505e9                            bin\win64\pgo\m_lxe.dll+27264489 QueryMLFcnTable_m_lxe+02236005
[ 27] 0x0000000019750d76                            bin\win64\pgo\m_lxe.dll+27266422 QueryMLFcnTable_m_lxe+02237938
[ 28] 0x0000000019751bf9                            bin\win64\pgo\m_lxe.dll+27270137 QueryMLFcnTable_m_lxe+02241653
[ 29] 0x00000000197518d2                            bin\win64\pgo\m_lxe.dll+27269330 QueryMLFcnTable_m_lxe+02240846
[ 30] 0x0000000019751832                            bin\win64\pgo\m_lxe.dll+27269170 QueryMLFcnTable_m_lxe+02240686
[ 31] 0x000000001975f225                            bin\win64\pgo\m_lxe.dll+27324965 QueryMLFcnTable_m_lxe+02296481
[ 32] 0x000000001975ea1c                            bin\win64\pgo\m_lxe.dll+27322908 QueryMLFcnTable_m_lxe+02294424
[ 33] 0x000000001975cf95                            bin\win64\pgo\m_lxe.dll+27316117 QueryMLFcnTable_m_lxe+02287633
[ 34] 0x00000000196e9c0e                            bin\win64\pgo\m_lxe.dll+26844174 QueryMLFcnTable_m_lxe+01815690
[ 35] 0x00000000196f32d2                            bin\win64\pgo\m_lxe.dll+26882770 QueryMLFcnTable_m_lxe+01854286
[ 36] 0x0000000019483cf6                            bin\win64\pgo\m_lxe.dll+24329462 mwboost::serialization::singleton<mwboost::archive::detail::extra_detail::guid_initializer<MathWorks::lxe::CodeClearEvent> >::get_mutable_instance+21875354
[ 37] 0x0000000017dbb265                            bin\win64\pgo\m_lxe.dll+00438885
[ 38] 0x0000000017dba88c                            bin\win64\pgo\m_lxe.dll+00436364
[ 39] 0x0000000017db8779                            bin\win64\pgo\m_lxe.dll+00427897
[ 40] 0x0000000017db90eb                            bin\win64\pgo\m_lxe.dll+00430315
[ 41] 0x0000000017db8a49                            bin\win64\pgo\m_lxe.dll+00428617
[ 42] 0x00000000170ae007                     bin\win64\pgo\m_dispatcher.dll+00057351 Mfh_file::dispatch_fh_impl+00001111
[ 43] 0x00000000170ada9e                     bin\win64\pgo\m_dispatcher.dll+00055966 Mfh_file::dispatch_fh+00000062
[ 44] 0x0000000017da4ead                            bin\win64\pgo\m_lxe.dll+00347821
[ 45] 0x0000000017f3e9b6                            bin\win64\pgo\m_lxe.dll+02025910 MathWorks::lxe::ShutdownLxeEngine+00004034
[ 46] 0x0000000017e9fd3c                            bin\win64\pgo\m_lxe.dll+01375548 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00493568
[ 47] 0x0000000017ea091c                            bin\win64\pgo\m_lxe.dll+01378588 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00496608
[ 48] 0x0000000017ea1c92                            bin\win64\pgo\m_lxe.dll+01383570 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00501590
[ 49] 0x0000000017ea28f8                            bin\win64\pgo\m_lxe.dll+01386744 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00504764
[ 50] 0x0000000017ea1ddf                            bin\win64\pgo\m_lxe.dll+01383903 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00501923
[ 51] 0x0000000017ea1ede                            bin\win64\pgo\m_lxe.dll+01384158 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00502178
[ 52] 0x0000000017da9a7d                            bin\win64\pgo\m_lxe.dll+00367229
[ 53] 0x0000000017dbb265                            bin\win64\pgo\m_lxe.dll+00438885
[ 54] 0x0000000017dba88c                            bin\win64\pgo\m_lxe.dll+00436364
[ 55] 0x0000000017db42a1                            bin\win64\pgo\m_lxe.dll+00410273
[ 56] 0x0000000017db39c6                            bin\win64\pgo\m_lxe.dll+00408006
[ 57] 0x0000000017db3ace                            bin\win64\pgo\m_lxe.dll+00408270
[ 58] 0x0000000017f93780                            bin\win64\pgo\m_lxe.dll+02373504 mwboost::archive::detail::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>+00012824
[ 59] 0x0000000017f9372e                            bin\win64\pgo\m_lxe.dll+02373422 mwboost::archive::detail::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>::oserializer<mwboost::archive::xml_oarchive,MathWorks::lxe::CodeClearEvent>+00012742
[ 60] 0x0000000017e734fd                            bin\win64\pgo\m_lxe.dll+01193213 mwboost::archive::detail::iserializer<mwboost::archive::binaryTerm_iarchive,MathWorks::lxe::function_descriptor>::load_object_data+00311233
[ 61] 0x00000000179b69e7                    bin\win64\pgo\m_interpreter.dll+00354791 inEvalCmdWithLocalReturn+00000063
[ 62] 0x00000000fb610388                          bin\win64\libmwbridge.dll+00131976 mnParser+00001304
[ 63] 0x0000000016de3618                                  bin\win64\mcr.dll+00341528 mcr::runtime::setInterpreterThreadSingletonToCurrent+00020808
[ 64] 0x0000000016de414d                                  bin\win64\mcr.dll+00344397 mcr::runtime::setInterpreterThreadSingletonToCurrent+00023677
[ 65] 0x0000000016dab1ba                                  bin\win64\mcr.dll+00111034 mcrOptions::set_use_license_manager+00075498
[ 66] 0x0000000016dc6e74                                  bin\win64\mcr.dll+00224884 mcrOptions::set_use_license_manager+00189348
[ 67] 0x00000000fd02cfee                                  bin\win64\iqm.dll+00643054 iqm::PackagedTaskPlugin::execute+00000894
[ 68] 0x00000000fd02ce63                                  bin\win64\iqm.dll+00642659 iqm::PackagedTaskPlugin::execute+00000499
[ 69] 0x00000000fcffcc07                                  bin\win64\iqm.dll+00445447 iqm::Iqm::setupIqmFcnPtrs+00076215
[ 70] 0x00000000fcffcc59                                  bin\win64\iqm.dll+00445529 iqm::Iqm::setupIqmFcnPtrs+00076297
[ 71] 0x00000000fcffca16                                  bin\win64\iqm.dll+00444950 iqm::Iqm::setupIqmFcnPtrs+00075718
[ 72] 0x00000000fcfd9748                                  bin\win64\iqm.dll+00300872 iqm::Iqm::deliver+00005880
[ 73] 0x00000000fcfd8eef                                  bin\win64\iqm.dll+00298735 iqm::Iqm::deliver+00003743
[ 74] 0x00000000fcfdac0d                                  bin\win64\iqm.dll+00306189 iqm::Iqm::deliver+00011197
[ 75] 0x0000000100203775                        bin\win64\libmwservices.dll+02176885 services::system_events::PpeDispatchHook::dispatchOne+00036213
[ 76] 0x0000000100212a63                        bin\win64\libmwservices.dll+02239075 sysq::addProcessPendingEventsUnitTestHook+00006035
[ 77] 0x0000000100214310                        bin\win64\libmwservices.dll+02245392 sysq::getCondition+00004208
[ 78] 0x000000010021596d                        bin\win64\libmwservices.dll+02251117 svWS_ProcessPendingEvents+00000221
[ 79] 0x0000000016de7724                                  bin\win64\mcr.dll+00358180 mcr::runtime::setInterpreterThreadSingletonToCurrent+00037460
[ 80] 0x0000000016de7f16                                  bin\win64\mcr.dll+00360214 mcr::runtime::setInterpreterThreadSingletonToCurrent+00039494
[ 81] 0x0000000016ddcabf                                  bin\win64\mcr.dll+00314047 mcr_process_events+00001007
[ 82] 0x0000000016ced040                             bin\win64\MVMLocal.dll+00380992 mvm_server::inproc::LocalFactory::terminate+00177312
[ 83] 0x00000000fa9d9480                                  bin\win64\mvm.dll+01741952 mvm::detail::SessionImpl::initWithOptions+00000592
[ 84] 0x00000000fa9da170                                  bin\win64\mvm.dll+01745264 mvm::detail::SessionImpl::runMain+00000128
[ 85] 0x00000000fa9da395                                  bin\win64\mvm.dll+01745813 mvm::detail::SessionImpl::runMatlabDesktop+00000261
[ 86] 0x0000000140007036                               bin\win64\MATLAB.exe+00028726 mwboost::serialization::singleton_module::unlock+00000966
[ 87] 0x0000000140007f13                               bin\win64\MATLAB.exe+00032531 mwboost::serialization::singleton_module::unlock+00004771
[ 88] 0x00007ff8b0467974                   C:\Windows\System32\KERNEL32.DLL+00096628 BaseThreadInitThunk+00000020
[ 89] 0x00007ff8b2e2a271                      C:\Windows\SYSTEM32\ntdll.dll+00434801 RtlUserThreadStart+00000033

This error was detected while a MEX-file was running. If the MEX-file
is not an official MathWorks function, please examine its source code
for errors. Please consult the External Interfaces Guide for information
on debugging MEX-files.

**Comment by marius10p** [comment]
Thanks John. Seems like something broke in the CUDA code. A few questions about your configuration first:

1) Did you compile the CUDA code with mex and have you made changes to your configuration since then? Can you try compiling it again?
2) What is your C++ compiler? Run mex -setup c++ to know for sure. 
3) Are you up to date on Nvidia drivers? 

Assuming it's not one of these, something in your data must be triggering a bug in the low-level CUDA function. It could be a batch with very few spikes, a memory leak or something like that. You can monitor GPU RAM useage in the task manager. Have you changed any of the default settings?

**Comment by johnhwen1** [comment]
Thanks for the suggestions, Marius! 

1. I'll try recompiling again and let you know how it works.
2. The C++ compiler is Microsoft Visual C++ 2013. I see though that I also have a MinGW64 Compiler. 
3. I tried updating the Nvidia drivers with no luck.

I'll try monitoring the GPU RAM usage next time I run kilosort and let you know what happens!

**Comment by marius10p** [comment]
Is the compiler the community edition or Pro? I vaguely remember hearing about problems on the Pro.

**Comment by johnhwen1** [comment]
The compiler is the community edition. I've also tried recompiling and running kilosort2 again, but sadly the computer still restarts!

**Comment by chenchals** [comment]
Here is how I have resolved CUDA issues with Mex compilation in Matlab. At a basic level the listed below are needed, apart from installing compatible C/C++ compiler ( *note* latest is not the greatest IMHO)
Mex compilation issues:
1. Check: [https://www.mathworks.com/help/parallel-computing/gpu-support-by-release.html]
2. Check CUDA driver compatibility: [https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html]
3. Check the driver version on your NVIDIA card: Run -> nvidia-smi | grep 'Driver Version'
4. Now ensure the following: The driver version installed on NVIDA card (step 3) matches the CUDA compatibility version (step 2) and that particular version of CUDA is compatible with the Matlab version (step 1) you are using

**Comment by johnhwen1** [comment]
Hi chenchals,

Thanks for your suggestions. 

I'm running MATLAB 2018b with CUDA toolkit 9.1 (they're compatible). This particular CUDA version requires at least driver version 391.29 or greater, and I currently have 430.86. So it seems like everything checks out...

**Comment by johnhwen1** [comment]
Hi all, thanks again for all the help. Just wanted to let you know that I tried using the MinGW64 compiler and kilosort was able to finish without crashing. I'll try again to make sure it wasn't a fluke. But just wondering, are there any problems associated with using the MinGW64 compiler?

**Comment by marius10p** [comment]
That's very interesting, I didn't think it would work. It wasn't working a few years ago. How do your runtimes compare? Does everything look good in Phy?

**Comment by johnhwen1** [comment]
Runtimes were comparable, but I wasn't able to run Phy unfortunately because a new params file wasn't generated.. 

I've also tried rerunning kilosort on the MinGW64 compiler, but kilosort crashed again. Trying again with the studio c++ compiler. Will keep you updated!

**Comment by marius10p** [comment]
Any updates on this?

**Comment by marius10p** [comment]
Closing, reopen if necessary.

**Comment by LJacinto** [comment]
Did anyone solve this issue? I am having the exact same problem. Can we reopen this?

Kilosort crashes (and the computer re-starts) everytime  after going through all the batches in the main optmization, just after the "ntot" variable is displayed on the command window.

I'm using:

Matlab 2018a
CUDA toolkit 9.0.76
Visual Studio Community 2013 update 5
NVIDIA GeForce  GTX1060 (6Gb) with the most recent driver 441.12 (have also tried 385.54 which is  listed as minimum requirement for CUDA toolkit 9.0.76)

and I run 'mexGPUall' before I run kilosort everytime. MEX compilation is always trouble-free.

Strangely, I was able to run Kilosort all the way through 2 times. But on all other attempts (even for the 2 files that ran successfully once) Kilosort crashes and the computer restarts.

Any thoughts?

**Comment by LJacinto** [comment]
Actually, I tried changing the CUDA_CACHE_MAXSIZE as suggested in another issue reporting CUDA breakdown and it did the trick. Kilosort is now able to run all the way through without crashing and the computer re-starting on its own.

It seems that if the CUDA cache size it's not sufficient computer RAM just "blows up" (I was monitoring computer resources) and the computer crashes interely. Can something along this lines provide an explanation to what might be going on with the reported crash and re-start?

Anyway, from https://github.com/cortex-lab/KiloSort/blob/master/Docs/readme_win_linux.txt

In order to fix this set an environment variable "CUDA_CACHE_MAXSIZE " on the machine to 
some high value like 1GB.  By default "CUDA_CACHE_MAXSIZE" is 32MB. 
In Windows you can do this in properties > advanced system settings > environment variables. 
In order to set the cache to 1GB use CUDA_CACHE_MAXSIZE 1073741824.

**Comment by teresajurado** [comment]
Hi all!
We are having problems with the GPU of two computers (computer 1 and 2, please see config below). After having successfully installed and used kilosort 2.5 both computers started to crash and reboot. Computer 1 started to reboot when Matlab was running a script using parallel computing toolbox and after this issue kilosort never worked again. Computer 2 started to reboot and now GPU doesn’t work any more and we will have to replace it. As suggested in this list, in computer 1, I set an environment variable "CUDA_CACHE_MAXSIZE " on the machine and set the cache to 1073741824. After doing this, kilosort did not fail again but when I tried to run the parallel computing toolbox  in Matlab I got this message:
Warning: A task was unexpectedly found in state: Unavailable.
 > In parallel.cluster.Local/deriveJobStateFromTasks (line 29)
   In parallel.cluster/Local/getJobState (line 174)
   In parallel.cluster/CJSCluster/hGetJobState (line 375)
   In parallel.internal.cluster/CJSJobMixin/hGetStateEnum (line 175)
   In parallel.Job.get.StateEnum (line 242)
   In parallel.Job.get.State (line 234)
   In parallel.internal.customattr.CustomGetSet>iVectorisedGetHelper (line 128)
   In parallel.internal.customattr.CustomGetSet>@(a,b,c)iVectorisedGetHelper(obj,a,b,c) (line 102)
   In parallel.internal.customattr/CustomGetSet/doVectorisedGet (line 103)
   In parallel.internal.customattr/CustomGetSet/hVectorisedGet (line 76)
   In parallel.internal.customattr.GetSetImpl>iAccessProperties (line 322)
   In parallel.internal.customattr.GetSetImpl>iGetAllPropertiesVec (line 264)
   In parallel.internal.customattr/GetSetImpl/getImpl (line 133)
   In parallel.internal.customattr.CustomGetSet>iHetFunGetFunction (line 154)
   In parallel.internal.customattr.CustomGetSet>@(o)iHetFunGetFunction(o,props) (line 139)
   In parallel.internal.cluster.hetfun (line 46)
   In parallel.internal.customattr.CustomGetSet>iHetFunGetProperty (line 139)
   In parallel.internal.customattr/CustomGetSet/get (line 38)
   In parallel.internal.pool/InteractiveClient/pRemoveOldJobs (line 468)
   In parallel.internal.pool/InteractiveClient/start (line 311)
   In parallel.internal.pool.AbstractClusterPool>iStartClient (line 826)
   In parallel.internal.pool/AbstractClusterPool/hBuildPool (line 596)
   In parallel.internal.pool.doParpool (line 20)
   In parpool (line 122)
   In parallel.internal.pool/PoolArrayManager/getOrAutoCreateWithCleanup (line 58)
   In pctTryCreatePoolIfNecessary (line 28)
   In parallel_function (line 418)
Because the parallel computing toolbox is using GPU, I am afraid that the GPU was somehow damaged while running kilosort 2.5 before I increased CUDA_CACHE_MAXSIZE.
Has anyone had the same problem? Do you think increasing CUDA CACHE SIZE is enough to protect the GPU?
Computer 1: NDIVIA GeForce RTX 3070 Ti, CUDA Toolkit version 10.1 and Matlab 2020a
Computer 2: NDIVIA GeForce GTX 1050 Ti, CUDA Toolkit version 10.0 and Matlab 2019a
5:07
Thank you!
